File: ios/EmojisPopupWrapperModule.swift
import ExpoModulesCore

public class EmojisPopupWrapperModule: Module {
  public func definition() -> ModuleDefinition {
    Name("EmojisPopupWrapper")

    View(ReactionPopupAnchorView.self) {
      Events([
        "onDragSelect",
        "onDragPlus",
        "onDragDismiss"
      ])

      Prop("anchorId") { (view: ReactionPopupAnchorView, anchorId: String) in
        view.anchorId = anchorId
      }

      Prop("gestureMode") { (view: ReactionPopupAnchorView, mode: String) in
        view.gestureMode = mode
      }

      Prop("dragParams") { (view: ReactionPopupAnchorView, params: [String: Any]?) in
        view.dragParams = params
      }
    }
  }
}

File: android/src/main/java/expo/modules/nativeemojispopup/EmojisPopupWrapperModule.kt
package expo.modules.nativeemojispopup

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class EmojisPopupWrapperModule : Module() {
  override fun definition() =
    ModuleDefinition {
      Name("EmojisPopupWrapper")

      View(EmojisPopupWrapper::class) {
        Events(arrayOf("onDragSelect", "onDragPlus", "onDragDismiss", "onTap"))

        Prop("anchorId") { view: EmojisPopupWrapper, anchorId: String ->
          view.anchorId = anchorId
        }

        Prop("gestureMode") { view: EmojisPopupWrapper, mode: String ->
          view.gestureMode = mode
        }

        Prop("dragParams") { view: EmojisPopupWrapper, params: Map<String, Any?>? ->
          view.dragParams = params
        }
      }
    }
}
